From 30e4a3436dee4c644667e5d2978072723452e5fe Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Sun, 23 Mar 2025 23:59:33 +0100
Subject: [PATCH] pinctrl: mediatek: disable MDC IO Master Mode on selecting
 MDIO

Disable MDC IO Master Mode on selecting MDIO function as it was notice
it does cause wrong MDIO bus detection.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/pinctrl/mediatek/pinctrl-airoha.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/pinctrl/mediatek/pinctrl-airoha.c b/drivers/pinctrl/mediatek/pinctrl-airoha.c
index 547a798b71c8..6e81760e31f6 100644
--- a/drivers/pinctrl/mediatek/pinctrl-airoha.c
+++ b/drivers/pinctrl/mediatek/pinctrl-airoha.c
@@ -746,7 +746,7 @@ static const struct airoha_pinctrl_func_group mdio_func_group[] = {
 			AIROHA_FUNC_MUX,
 			REG_GPIO_2ND_I2C_MODE,
 			GPIO_MDC_IO_MASTER_MODE_MODE,
-			GPIO_MDC_IO_MASTER_MODE_MODE
+			0
 		},
 		.regmap_size = 2,
 	},
-- 
2.48.1

